std/file.d

```diff
@@ -5119,3 +5120,77 @@ string tempDir() @trusted
     myFile.write("hello");
     assert(myFile.readText == "hello");
 }
+
+/**
+* Returns the available space in a file system.
+*
+* On `Windows`, this function uses the result of the Windows API function
+* $(LINK2 https://msdn.microsoft.com/en-us/library/windows/desktop/aa364937(v=vs.85).aspx, $(D GetDiskFreeSpaceExW)).
+*
+* On `POSIX` platforms, it uses the $(LINK2 http://man7.org/linux/man-pages/man3/statvfs.3.html,
+* $(D statvfs)) function.
+*
+* Due to inconsistent behaviour with the `statvfs` `Posix` function,
+* `FreeBSD` does not support `getAvailableDiskSpace`.
+*
+* Params:
+*     path = a valid path to any file in the file system
+*
+* Returns:
+*    ulong which contains the available space in bytes
+*
+* Throws:
+*     $(LREF FileException) in case statvfs (`POSIX`) or GetDiskFreeSpaceExW (`Windows`) fails.
+*/
+ulong getAvailableDiskSpace(string path) @trusted
+{
+    import std.exception : enforce;
+    import std.conv : text;
+
+    version (FreeBSD)
+    {
+        static assert(0, "Unsupported platform");
```

Comment:
FYI: You could put the `extern(C)` declarations from your druntime PR here inline.
It seems like getting D to work with FreeBSD12 is a lot of work anyways (and not supported at the moment) and in this way `getAvailableDiskSpace` at least supports "all currently supported versions" of FreeBSD and this can be tweaked later once a solution to the druntime PR is found.

Reply:
Please no. We have sometimes put `extern(C)` declarations in Phobos, but in general, they should go in druntime, and every one of them that ends up in Phobos, it's something that has to be fixed later. And in the case of FreeBSD with regards to 11 vs 12, having `extern(C)` declarations in Phobos is definitely going to make life harder. If there is a missing C declaration in druntime, just add it in druntime first, and if there's a PR for that, please wait on it rather than adding the declaration here to push this PR through. Doing that is just going to create more problems later.